das/daser.go

```diff
@@ -70,18 +77,20 @@ func (d *DASer) Start(context.Context) error {
 	}
 	log.Infow("loaded checkpoint", "height", checkpoint)
 
-	dasCtx, cancel := context.WithCancel(context.Background())
-	d.cancel = cancel
+	d.ctx, d.cancel = context.WithCancel(context.Background())
 
 	// kick off catch-up routine manager
-	go d.catchUpManager(dasCtx, checkpoint)
+	go d.catchUpManager(d.ctx, checkpoint)
 	// kick off sampling routine for recently received headers
-	go d.sample(dasCtx, sub, checkpoint)
+	go d.sample(d.ctx, sub, checkpoint)
 	return nil
 }
 
 // Stop stops sampling.
 func (d *DASer) Stop(ctx context.Context) error {
+	if d.ctx.Err() != nil {
+		return nil
+	}
```

Comment:
Is it okay to swallow this error silently? If yes then you should comment why, since it's not immediately obvious.

Reply:
added.